app/routes/workshops/index.tsx

```diff
@@ -27,26 +34,134 @@ export function meta() {
   }
 }
 
+function truncate(text: string, length: number) {
+  if (!text || text.length <= length) {
+    return text
+  }
+
+  return `${text.substr(0, length).trim()}…`
+}
+
+function WorkshopCard({
+  slug,
+  frontmatter: {
+    date,
+    title = 'Untitled Workshop',
+    description = 'Description TBA',
+    tech,
+  },
+}: MdxListItem) {
+  return (
+    <Link
+      to={slug}
+      className="focus-ring block flex flex-col p-16 pr-24 w-full h-full bg-gray-100 dark:bg-gray-800 rounded-lg"
+    >
+      <div className="flex-none h-36">
+        {/* TODO: how to determine if it's open or not? */}
+        {Math.random() * 10 < 5 ? (
+          <div className="inline-flex items-baseline">
+            <div className="block flex-none w-3 h-3 bg-green-600 rounded-full" />
+            <H6 as="p" className="pl-4">
+              Open for registration
+            </H6>
+          </div>
+        ) : null}
+      </div>
+      <div className="flex-none">
+        <div className="inline-block mb-4 px-8 py-4 text-black dark:text-white text-lg dark:bg-gray-600 bg-white rounded-full">
+          {tech}
+        </div>
+      </div>
+      <H3 className="flex-none mb-3">{title}</H3>
+      {/*TODO: line-clamp has a browser support of 90%, is that enough?*/}
```

Comment:
I think it's enough, especially since the target audience (developers) is usually using recent and up-to-date browsers. Besides, it isn't that it looks totally off without line-clamp either.